cloud/openstack/os_quota.py

```diff
@@ -0,0 +1,438 @@
+        per_volume_gigabytes=dict(required=False, type='int', default=None),
+        port=dict(required=False, type='int', default=None),
+        project=dict(required=False, type='int', default=None),
+        properties=dict(required=False, type='int', default=None),
+        ram=dict(required=False, type='int', default=None),
+        rbac_policy=dict(required=False, type='int', default=None),
+        router=dict(required=False, type='int', default=None),
+        secgroup_rules=dict(required=False, type='int', default=None),
+        secgroups=dict(required=False, type='int', default=None),
+        server_group_members=dict(required=False, type='int', default=None),
+        server_groups=dict(required=False, type='int', default=None),
+        snapshots=dict(required=False, type='int', default=None),
+        snapshots_types=dict(required=False, type='dict', default={}),
+        subnet=dict(required=False, type='int', default=None),
+        subnetpool=dict(required=False, type='int', default=None),
+        volumes=dict(required=False, type='int', default=None),
+        volumes_types=dict(required=False, type='dict', default={})
+    )
+
+    module = AnsibleModule(argument_spec,
+            supports_check_mode=True
+        )
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    try:
+        cloud_params = dict(module.params)
+        cloud = shade.operator_cloud(**cloud_params)
+
+        #In order to handle the different volume types we update module params after.
+        dynamic_types = [
+            'gigabytes_types',
+            'snapshots_types',
+            'volumes_types',
+        ]
+
+        for dynamic_type in dynamic_types:
+            for k, v in module.params[dynamic_type].iteritems():
+                module.params[k] = int(v)
+
+        #Get current quota values
+        project_quota_output = _get_quotas(cloud, cloud_params['name'])
+
+        if module.check_mode:
+            module.exit_json(changed=_system_state_change(module, project_quota_output))
+
+        changes_required, quota_change_request = _system_state_change_details(
+                module,
+                project_quota_output
+            )
+
+        if changes_required:
+            for quota_type in quota_change_request.keys():
+                quota_call = getattr(cloud, 'set_%s_quotas' % (quota_type))
+                quota_call(cloud_params['name'], **quota_change_request[quota_type])
+
+            #Get quota state post changes for validation
+            project_quota_update = _get_quotas(cloud, cloud_params['name'])
+
+            if project_quota_output == project_quota_update:
+                module.fail_json(msg='Could not apply quota update')
+
+            project_quota_output = project_quota_update
+
+        module.exit_json(changed=changes_required, ansible_facts=dict(
+            openstack_quotas=project_quota_output)
```

Comment:
I would recommend what you suggest. Remove `ansible_facts` and just have `openstack_quotas`.  That should match the written documentation.